packages/flame/lib/src/game/mixins/has_collidables.dart

```diff
@@ -29,6 +19,6 @@ mixin HasCollidables on FlameGame {
   }
 
   void handleCollidables() {
-    collisionDetection(collidables);
+    collisionDetection(children.query<Collidable>());
```

Comment:
Isn't the issue here that we need to be able to handle collisions between components that are in separate parts of the component tree? `children.query` will not traverse the component tree, primarily because it doesn't know that components are arranged into a tree; but even if it did know that -- it would still be inefficient to re-create the list of collidables on every tick.

Reply:
I think Pasha has a good point here, I am almost sure that if we remove this internal tracking of components here, we will undo a task that @spydon recently did, so the component children are considered on the collisions detections.

Maybe for this fix, we should just keep the addition of the must call super on onLoad and revert this change?

Reply:
ah, I see, didn't think of that case. but I feel that normally we do these things using `propagateToChildren`, instead of keeping track of ad-hoc lists, right?

Reply:
> ah, I see, didn't think of that case. but I feel that normally we do these things using `propagateToChildren`, instead of keeping track of ad-hoc lists, right?

That will be less performant, and we have done it like that on the things where the parent covers the child, but apparently people don't necessarily use the Collidables like that (from the issues we have seen on discord). The best thing would be if we had a method to query a tree like we have the other flat query, then we could also use that for gestures too.

Reply:
that would be a great addition I think! for now I think I will bring back the collidables and just make sure remove is called properly; we can iterate later on a generic recursive `query` method

Reply:
`propagateToChildren` would not be ideal here, since we need to iterate over every pair of elements in the list.